### PR TITLE
feat(web): Add album sorting to albums view

### PR DESCRIPTION
**Pull request by klejejs**
The feature adds the option to sort albums on the web. The inspiration for sorting options comes from Google Photos.
Visually it looks like this:
image
image

### CODE REVIEW

File: mobile/openapi/lib/model/album_response_dto.dart [hunk 1]
```diff
@@ -48,6 +49,14 @@ class AlbumResponseDto {
...
   UserResponseDto owner;
 
+  ///
+  /// Please note: This property should have been non-nullable! Since the specification file
+  /// does not include a default value (using the "default:" property), however, the generated
+  /// source code must fall back to having a nullable type.
+  /// Consider adding a "default:" property in the specification file to hide this note.
+  ///
+  DateTime? lastModifiedAssetTimestamp;
+
```
Comment (klejejs) [inline mobile/openapi/lib/model/album_response_dto.dart:59]: Was there a better way to do this to avoid the message?

Comment (alextran1502) [reply mobile/openapi/lib/model/album_response_dto.dart:59]: This should be OK, we have other DTOs with properties that have the same warning as well.

**Comment by klejejs** [comment]
I am running into an issue where the linter tells me to use tabs, but the prettier changes the tab to 2 spaces. Any ideas on how to fix that?

**Comment by alextran1502** [comment]
> I am running into an issue where the linter tells me to use tabs, but the prettier changes the tab to 2 spaces. Any ideas on how to fix that?

@jrasm91 Can you help explain this tab war?

**Comment by jrasm91** [comment]
Not sure. You can run `npm run format:fix` though in both the server and web folders. The server uses spaces and the web uses tabs I believe.

**Comment by jrasm91** [comment]
The actual error is _mixed_ spaces and tabs. Just use tabs and you should be fine

**Comment by alextran1502** [comment]
Would it be possible to perform this type of sorting on the front end only?

**Comment by klejejs** [comment]
> Would it be possible to perform this type of sorting on the front end only?

To do that, we would need to return all photos for all albums when they are loaded. If there are many albums and many photos in them, I worry that it might take time to find the most recent photo for each one. Furthermore, people with a slow internet connection would experience delays as a lot of data would need to be loaded at once.

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**